python/cudf/cudf/core/column/column.py

```diff
@@ -15,32 +15,186 @@
 import cudf
 import cudf._lib as libcudf
 from cudf._lib.stream_compaction import nunique as cpp_unique_count
+from cudf._libxx.column import Column
 from cudf.core.buffer import Buffer
+from cudf.core.dtypes import CategoricalDtype
 from cudf.utils import cudautils, ioutils, utils
-from cudf.utils.dtypes import is_categorical_dtype, is_scalar, np_to_pa_dtype
-from cudf.utils.utils import buffers_from_pyarrow
+from cudf.utils.dtypes import (
+    is_categorical_dtype,
+    is_scalar,
+    is_string_dtype,
+    np_to_pa_dtype,
+)
+from cudf.utils.utils import (
+    buffers_from_pyarrow,
+    calc_chunk_size,
+    mask_bitsize,
+)
+
+
+class ColumnBase(Column):
+    def __init__(
+        self, data, size, dtype, mask=None, offset=0, name=None, children=()
+    ):
+        """
+        Parameters
+        ----------
+        data : Buffer
+        dtype
+            The type associated with the data Buffer
+        mask : Buffer, optional
+        name : optional
+        children : tuple, optional
+        """
 
+        super().__init__(
+            data,
+            size=size,
+            dtype=dtype,
+            mask=mask,
+            offset=offset,
+            children=children,
+            name=name,
+        )
+        self._data = data
+        self.size = size
+        self.dtype = dtype
+        self._mask = mask
+        self.offset = offset
+        self.children = children
+        self.name = name
+
+    def __reduce__(self):
+        return (
+            build_column,
+            (
+                self.data,
+                self.dtype,
+                self.mask,
+                self.offset,
+                self.children,
+                self.name,
+            ),
+        )
 
-class Column(object):
-    """An immutable structure for storing data and mask for a column.
-    This should be considered as the physical layer that provides
-    container operations on the data and mask.
+    @property
+    def data(self):
+        return self._data
 
-    These operations work on each data element as plain-old-data.
-    Any logical operations are implemented in subclasses of *TypedColumnBase*.
+    @data.setter
+    def data(self, value):
+        self._data = value
 
-    Attributes
-    ----------
-    _data : Buffer
-        The data buffer
-    _mask : Buffer
-        The validity mask
-    _null_count : int
-        Number of null values in the mask.
-
-    These attributes are exported in the properties (e.g. *data*, *mask*,
-    *null_count*).
-    """
+    @property
+    def mask(self):
+        return self._mask
+
+    @mask.setter
+    def mask(self, value):
+        self._nvstrings = None
```

Comment:
If we have a string column and someone sets the mask this looks like it will blow things up?